### DISCUSSION TITLE
Can I get a double dropdown in Storybook Controls addon?

### DISCUSSION THREAD

**Discussion by annacv** [storybookjs/storybook#12827] [Help]
I'm migrating from Knobs to Controls and I'm not being able to use a double dropdown selector to display a prop's value. I reached this in knobs:
`const iconClass = options('Type of Spacing', UI_SPACING, 'mr-', { display: 'select' }) + options('Num. of Spacing', UI_SIZE_NUMBERS, '1', { display: 'select' })`

This is not about using a 'multi-select' argTypes, it's about having one argTypes with more than one selector. Has anyone tried this? Thks!

**Comment by shilman** [comment]
You'd need to define custom argTypes for each dropdown:

```
export default {
  ....
  argTypes: {
    spacingType: {
      control: { type: 'select', options: [ .... ] }
    },
    spacingCount: {
      control: { type: 'select', options: [ .... ] }
    },
  },
}
```

Then you'd combine those in the story:

```
export const SomeStory = ({ spacingType, spacingCount, ...rest }) => {
  const iconClass = f(spacingType, spacingCount) // however you combine them
  return <YourComponent iconClass={iconClass} {...rest} />;
}
```

This is a rough sketch but hope it makes sense. More on custom argTypes:

https://storybook.js.org/docs/react/essentials/controls#fully-custom-args

cc @jonniebigodes

**Reply by annacv** [reply]
It helped a lot! Thks!
